nixos/tests/matrix-appservice-irc.nix

```diff
@@ -0,0 +1,155 @@
+import ./make-test-python.nix ({ pkgs, ... }:
+  let
+    homeserverUrl = "http://homeserver:8448";
+  in
+  {
+
+    name = "matrix-synapse";
+    meta = {
+      maintainers = pkgs.matrix-appservice-irc.meta.maintainers;
+    };
+
+    nodes = {
+      homeserver = { pkgs, ... }: {
+        # We'll switch to this once the config is copied into place
+        specialisation.running.configuration = {
+          services.matrix-synapse = {
+            enable = true;
+            database_type = "sqlite3";
+            app_service_config_files = [ "/registration.yml" ];
```

Comment:
How does the absolute path here work?

Reply:
Oh, the registration file is copied to this location. Weird choice of location.

Reply:
I wonder why synapse should even be allowed to read this location. Maybe the synapse home would be a more appropriate location, and then we think about locking down synapse more in a next step.